lib/streamlit/config.py

```diff
@@ -202,6 +202,19 @@ def _delete_option(key):
 )
 
 
+_create_option(
+    "global.disableWatchdog",
+    description="""
+        By default, Streamlit detects file changes with the Python watchdog
+        module. Optionally, you can force Streamlit to use Polling even if 
+        watchdog is available. This is not recommended.
+
+        If you'd like to use polling, set this to True.
+    """,
+    default_val=False,
+    type_=bool,
+)
```

Comment:
`disableWatchdog` makes it seem like we won't watch files at all. I think the right API here would be something like: `fileWatcherType = "none" | "watchdog" | "poll"`

Reply:
That's a good point. So maybe: `fileWatcherType = "none" | "auto" | "watchdog" | "poll"`. And only `auto` has the fail-over.

Ugggh... this PR is getting larger. Sorry!